pylutron/__init__.py

```diff
@@ -655,6 +694,496 @@ def handle_update(self, args):
+            return False
+
+        def _u_operating_mode(mode):
+            """Handles operating mode interaction"""
+            if self._current_mode != HVAC.OperatingModes(int(mode)).name:
+              self._current_mode = HVAC.OperatingModes(int(mode)).name
+              self._query_waiters.notify()
+              self._dispatch_event(HVAC.Event.OPERATING_MODE, {'current_mode': self._current_mode})
+              return True
+            return False
+
+        def _u_fan_mode(mode):
+            """Handles fan mode interaction"""
+            if self._current_fan != HVAC.FanModes(int(mode)).name:
+              self._current_fan = HVAC.FanModes(int(mode)).name
+              self._query_waiters.notify()
+              self._dispatch_event(HVAC.Event.FAN_MODE, {'current_fan': self._current_fan})
+              return True
+            return False
+
+        def _u_call_status(mode):
+            """Handles call status"""
+            if self._call_status != HVAC.CallStatus(int(mode)).name:
+              self._call_status = HVAC.CallStatus(int(mode)).name
+              self._query_waiters.notify()
+              self._dispatch_event(HVAC.Event.CALL_STATUS, {'call_status': self._call_status})
+              return True
+            return False
+        
+        def _u_schedule_status(mode):
+            """Handles schedule status"""
+            if self._schedule_status != HVAC.ScheduleStatus(int(mode)).name:
+              self._schedule_status = HVAC.ScheduleStatus(int(mode)).name
+              self._query_waiters.notify()
+              self._dispatch_event(HVAC.Event.SCHEDULE_STATUS, {'schedule_status': self._schedule_status})
+              return True
+            return False
+
+        def _u_eco_mode(mode):
+            """Handles eco mode status"""
+            mode = int(mode)-1
+            if self._eco_mode != bool(mode):
+              self._eco_mode = bool(mode)
+              self._query_waiters.notify()
+              self._dispatch_event(HVAC.Event.ECO_MODE, {'eco_mode': self._eco_mode})
+              return True
+            return False
+
+        try:
+          event = HVAC.Event(int(args[0]))
+        except ValueError:
+          _LOGGER.warning('Lutron HVAC Action Number %s is not implemented', args[0])
+          return False  
+
+        handler_functions = {
+          HVAC.Event.TEMP_CURRENT_F: (_u_current_temp_f, 1),
+          HVAC.Event.TEMP_SETPOINTS_F: (_u_setpoints_f, 2),
+          HVAC.Event.OPERATING_MODE: (_u_operating_mode, 1),
+          HVAC.Event.FAN_MODE: (_u_fan_mode, 1),
+          HVAC.Event.CALL_STATUS: (_u_call_status,1),
+          HVAC.Event.TEMP_CURRENT_C: (_u_current_temp_c, 1),
+          HVAC.Event.TEMP_SETPOINTS_C: (_u_setpoints_c, 2),
+          HVAC.Event.SCHEDULE_STATUS: (_u_schedule_status, 1),
+          HVAC.Event.ECO_MODE: (_u_eco_mode, 1),
+        }
+        if event in handler_functions:
+            handler, num_args = handler_functions[event]
```

Comment:
Why not just pass all remaining arguments?

Reply:
I'm a little new to python so I'm not sure how to do that. If you could suggest an approach so I could learn that would be great.

Reply:
You'd replace this all with:

handler(*args[1:])